Core/Core/API/APIAssignmentRequestable.swift

```diff
@@ -88,28 +88,47 @@ public struct GetAssignmentsRequest: APIRequestable {
     public typealias Response = [APIAssignment]
 
     let courseID: String
+    let assignmentGroupID: String?
     let orderBy: OrderBy
+    let assignmentIDs: [String]?
     let include: [Include]
-    let querySize: Int
-
-    public init(courseID: String, orderBy: OrderBy = .position, include: [Include] = [], querySize: Int = 100) {
+    let perPage: Int?
+
+    public init(
+        courseID: String,
+        assignmentGroupID: String? = nil,
+        orderBy: OrderBy = .position,
+        assignmentIDs: [String]? = nil,
+        include: [Include] = [],
+        perPage: Int? = nil
+    ) {
         self.courseID = courseID
+        self.assignmentGroupID = assignmentGroupID
         self.orderBy = orderBy
+        self.assignmentIDs = assignmentIDs
         self.include = include
-        self.querySize = querySize
+        self.perPage = perPage
     }
 
     public var path: String {
         let context = ContextModel(.course, id: courseID)
+        if let assignmentGroupID = assignmentGroupID {
+            return "\(context.pathComponent)/assignment_groups/\(assignmentGroupID)/assignments"
+        }
         return "\(context.pathComponent)/assignments"
     }
 
     public var query: [APIQueryItem] {
-        var q: [APIQueryItem] = [ .value("order_by", orderBy.rawValue), .value("per_page", String(querySize)) ]
-
-        if !include.isEmpty {
-            q.append( .array("include", include.map { $0.rawValue }) )
+        var query: [APIQueryItem] = [
+            .value("order_by", orderBy.rawValue),
+            .array("include", include.map { $0.rawValue }),
+        ]
+        if let assignmentIDs = assignmentIDs {
+            query.append(.array("assignment_ids", assignmentIDs))
```

Comment:
could you ever have too many assignment ID's?

Reply:
Yeah if there are more than `per_page` or the default then it doesn't work. I ended up not using `assignment_id`s though because it's broken for observers. 😞